Frameworks/Foundation/NSArray.mm

```diff
@@ -947,21 +923,69 @@ + (NSArray*)arrayWithContentsOfURL:(NSURL*)aURL {
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (NSString*)descriptionWithLocale:(id)locale {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self descriptionWithLocale:locale indent:0];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (NSString*)descriptionWithLocale:(id)locale indent:(NSUInteger)level {
-    UNIMPLEMENTED();
-    return StubReturn();
+    thread_local unsigned int indent = 0;
+    NSMutableString* s = [NSMutableString string];
+    NSString* indentStr = (level == 0) ? @"    " : @"\t";
```

Comment:
Nope, level is different from our thread_local indent, it's a parameter to the method.  0 means use 4 spaces, 1 mean use a tab.  https://developer.apple.com/reference/foundation/nsarray/1416257-descriptionwithlocale